pkg/relayer/relay_authenticator/relay_signer.go

```diff
@@ -0,0 +1,33 @@
+package relay_authenticator
+
+import (
+	"github.com/pokt-network/poktroll/pkg/signer"
+	"github.com/pokt-network/poktroll/x/service/types"
+)
+
+// SignRelayResponse signs the hash of a RelayResponse given the supplier operator address.
+// It uses the keyring and keyName to sign the payload and returns the signature.
+func (ra *relayAuthenticator) SignRelayResponse(relayResponse *types.RelayResponse, supplierOperatorAddr string) error {
+	// create a simple signer for the request
+	operatorKeyName, ok := ra.operatorAddressToSigningKeyNameMap[supplierOperatorAddr]
+	if !ok {
+		return ErrRelayAuthenticatorUndefinedSigningKeyNames.Wrapf("unable to resolve the signing key name for %s", supplierOperatorAddr)
+	}
+	signer := signer.NewSimpleSigner(ra.keyring, operatorKeyName)
+
+	// extract and hash the relay response's signable bytes
+	signableBz, err := relayResponse.GetSignableBytesHash()
+	if err != nil {
+		return ErrRelayAuthenticatorInvalidRelayResponse.Wrapf("error getting signable bytes: %v", err)
+	}
+
+	// sign the relay response
+	responseSig, err := signer.Sign(signableBz)
+	if err != nil {
+		return ErrRelayAuthenticatorInvalidRelayResponse.Wrapf("error signing relay response: %v", err)
+	}
+
+	// set the relay response's signature
+	relayResponse.Meta.SupplierOperatorSignature = responseSig
```

Comment:
@red-0ne Per Bryan's comment in the other PR about "signing something with a nil field", that's what's happening here too.

Reply:
I might be missing context here but I believe we are doing it the right way:
* `RelayResponse.Meta` is a value type not a pointer
* `GetSignableBytesHash` is working with a value type not a pointer
* The signature field is intentionally cleared for hashing
* The signature is applied after computing the hash

If `RelayResponse.Meta.SessionHeader` which is a pointer is an issue, I'll leave a TODO_UPNEXT to address it as it might be a large change with a broader scope.

Reply:
There's no issue. This is an example that's similar to this

https://github.com/pokt-network/poktroll/pull/1075#discussion_r1961865859

![Screenshot 2025-02-20 at 3 44 40 PM](https://github.com/user-attachments/assets/8b6ab8a0-980d-4762-ae0f-e325df8c2dd8)